ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+int MMAddPointRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                           struct MiraMonFeature *hMMFeature,
+                           MM_INTERNAL_FID nElemCount)
+{
+    struct MM_DATA_BASE_XP *pBD_XP = nullptr;
+    MM_EXT_DBF_N_FIELDS nNumPrivateMMField = MM_PRIVATE_POINT_DB_FIELDS;
+    char *pszRecordOnCourse;
+    struct MM_FLUSH_INFO *pFlushRecList;
+
+    if (!hMiraMonLayer)
+        return 1;
+
+    // In V1.1 only _UI32_MAX records number is allowed
+    if (MMCheckVersionForFID(hMiraMonLayer,
+                             hMiraMonLayer->MMPoint.MMAdmDB.pMMBDXP->nRecords +
+                                 hMMFeature->nNumMRecords))
+        return MM_STOP_WRITING_FEATURES;
+
+    // Adding record to the MiraMon table (extended DBF)
+    // Flush settings
+    pFlushRecList = &hMiraMonLayer->MMPoint.MMAdmDB.FlushRecList;
+    pFlushRecList->pBlockWhereToSaveOrRead =
+        (void *)hMiraMonLayer->MMPoint.MMAdmDB.pRecList;
+
+    pBD_XP = hMiraMonLayer->MMPoint.MMAdmDB.pMMBDXP;
+
+    // Test lenght
+    if (MMDetectAndFixDBFWidthChange(hMiraMonLayer, hMMFeature,
+                                     &hMiraMonLayer->MMPoint.MMAdmDB,
+                                     pFlushRecList, nNumPrivateMMField, 0, 0))
+        return MM_FATAL_ERROR_WRITING_FEATURES;
+
+    // Reassign the point because the function can realloc it.
+    pszRecordOnCourse = hMiraMonLayer->MMPoint.MMAdmDB.szRecordOnCourse;
+    pFlushRecList->pBlockToBeSaved = (void *)pszRecordOnCourse;
+
+    // Now lenght is sure, write
+    memset(pszRecordOnCourse, 0, pBD_XP->BytesPerRecord);
+    MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse, pBD_XP->pField,
+                             &nElemCount, TRUE);
+
+    pFlushRecList->SizeOfBlockToBeSaved = pBD_XP->BytesPerRecord;
+    if (MMAddFeatureRecordToMMDB(
+            hMiraMonLayer, hMMFeature, &hMiraMonLayer->MMPoint.MMAdmDB,
+            pszRecordOnCourse, pFlushRecList,
+            &hMiraMonLayer->MMPoint.MMAdmDB.pMMBDXP->nRecords,
+            nNumPrivateMMField))
+        return MM_FATAL_ERROR_WRITING_FEATURES;
+    return MM_CONTINUE_WRITING_FEATURES;
+}
+
+// Adds a stringline record to a MiraMon table associated with a vector layer.
+int MMAddArcRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         struct MiraMonFeature *hMMFeature,
+                         MM_INTERNAL_FID nElemCount, struct MM_AH *pArcHeader)
+{
+    struct MM_DATA_BASE_XP *pBD_XP = nullptr;
+    char *pszRecordOnCourse;
+    struct MiraMonArcLayer *pMMArcLayer;
+    MM_EXT_DBF_N_FIELDS nNumPrivateMMField = MM_PRIVATE_ARC_DB_FIELDS;
+    struct MM_FLUSH_INFO *pFlushRecList;
+
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (hMiraMonLayer->bIsPolygon)
+        pMMArcLayer = &hMiraMonLayer->MMPolygon.MMArc;
+    else
+        pMMArcLayer = &hMiraMonLayer->MMArc;
+
+    if (!pMMArcLayer)
+        return 1;
+
```

Comment:
As you are taking above the address of a member of a structure, the resulting pointer cannot be null, so below test is useless:

```suggestion
```

Reply:
Indeed. Done.